### DISCUSSION TITLE
Filtering duplicate spans

### DISCUSSION THREAD

**Discussion by ricardokirkner** [jaegertracing/jaeger#4894] [Q&A] [answered]
I'm trying to understand the whole processing pipeline for opentelemetry and jaeger. I'm currently ingesting OTel traces using the jager-all-in-one image, just doing it like `curl --json @trace.json http://localhost:4318/v1/traces`.

The trace is ingested properly and I can see the trace/spans in the jaeger UI just fine.
The problem starts when I ingest the same trace again, in this case jaeger just happily accepts it again and the UI will show twice the amount of spans for the given trace (and the duplicate spans will have a warning about them being duplicate).

Is it possible to configure jaeger in a way to either (ideally) not store the duplicate spans, or at least not display them in the UI?

**Comment by yurishkuro** [comment, marked as answer]
Our other storage backends use primary keys based on span ID, so they should naturally dedupe identical spans (although to be confirmed if that's the case for Elasticsearch). The in-memory storage does not implement such de-duplication.

Overall, it seems like a reasonable function to have to dedupe identical spans on read. A bit of a caveat there is that Zipkin's legacy model assumes the same span ID between client and server, so we already had some logic to handle that.

**Reply by ricardokirkner** [reply]
Thank you so much for your answer.. this indeed was the case. I switched to the `badger` storage (even the ephemeral one) and I no longer have this issue.
